# Copyright 2026 Groundwork contributors.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# F1. The default uncaught-exception handler prints exactly what the JDK's ThreadGroup does:
# 'Exception in thread "main" ' + printStackTrace(System.err).
#
# Applies to: github.com/konsoletyper/teavm tag 0.13.1 (b3a245b7)

diff --git a/classlib/src/main/java/org/teavm/classlib/java/lang/TDefaultUncaughtExceptionHandler.java b/classlib/src/main/java/org/teavm/classlib/java/lang/TDefaultUncaughtExceptionHandler.java
index e0e2d7c..0e00ff8 100644
--- a/classlib/src/main/java/org/teavm/classlib/java/lang/TDefaultUncaughtExceptionHandler.java
+++ b/classlib/src/main/java/org/teavm/classlib/java/lang/TDefaultUncaughtExceptionHandler.java
@@ -22,8 +22,10 @@ public class TDefaultUncaughtExceptionHandler implements UncaughtExceptionHandle
     @Override
     public void uncaughtException(TThread t, Throwable e) {
 
-        // print to browser console by default
-        e.printStackTrace();
+        // teavm-fork: same output as the JDK's ThreadGroup.uncaughtException, so the first line is
+        // exactly 'Exception in thread "main" java.lang.ArithmeticException: / by zero'.
+        System.err.print("Exception in thread \"" + t.getName() + "\" ");
+        e.printStackTrace(System.err);
     }
 
 }
